packages/gotrue/lib/src/types/user_attributes.dart

```diff
@@ -35,6 +38,26 @@ class UserAttributes {
       if (data != null) 'data': data,
     };
   }
+
+  @override
+  bool operator ==(covariant UserAttributes other) {
```

Comment:
I don't like the covariant keyword here. It should be possible to compare any Object with this object, like when the comparing object is of type `dynamic` at compile time, this shouldn't throw an error at runtime.